### PR TITLE
[RELAY] Move frontend utils

### PR DESCRIPTION
**Pull request by mbaret**
The util file currently under frontend is used from outside of frontend (in qnn/op/legalizations). This suggests that the file should be pushed up to a higher level.

The benefit from this change is that importing qnn no longer also imports all the frontends. I've run into this trying to import qnn from within op.contrib.

### CODE REVIEW

**Comment by mbaret** [comment]
cc @anijain2305 @zhiics

**Comment by mbaret** [comment]
Does anyone have a chance to take a look at this?

**Comment by zhiics** [comment]
I am actually not quite encourage us to introduce this to the relay namespace as this file only really has one function and it is rarely used

**Comment by Somisary** [comment]
Where would be appropriate? An alternative would be that given this is used only in two places we could inline it into those files.

**Comment by anijain2305** [comment]
Maybe we want to move it some other already existing file. But, I think we need this in Relay namespace. This is used in TFLite parser and QNN Op legalization.

~~~
python/tvm/relay/frontend/tflite.py:from .util import get_scalar_from_constant
python/tvm/relay/frontend/tflite.py:        lhs_scale_value = get_scalar_from_constant(lhs_scale)
python/tvm/relay/frontend/tflite.py:        rhs_scale_value = get_scalar_from_constant(rhs_scale)
python/tvm/relay/frontend/tflite.py:        lhs_zero_point_value = get_scalar_from_constant(lhs_zero_point)
python/tvm/relay/frontend/tflite.py:        rhs_zero_point_value = get_scalar_from_constant(rhs_zero_point)
python/tvm/relay/frontend/tflite.py:            data_scale_val = get_scalar_from_constant(data_scale)
python/tvm/relay/frontend/tflite.py:            weight_scale_val = get_scalar_from_constant(weight_scale)
python/tvm/relay/frontend/tflite.py:                    pad_value = get_scalar_from_constant(input_tensor.qnn_params['zero_point'])
python/tvm/relay/frontend/tflite.py:            data_scale_val = get_scalar_from_constant(data_scale)
python/tvm/relay/frontend/tflite.py:            weight_scale_val = get_scalar_from_constant(weight_scale)
python/tvm/relay/frontend/util.py:def get_scalar_from_constant(expr):
python/tvm/relay/qnn/op/legalizations.py:from ...frontend.util import get_scalar_from_constant
python/tvm/relay/qnn/op/legalizations.py:        zero_point_val = get_scalar_from_constant(zero_point)
python/tvm/relay/qnn/op/legalizations.py:        zero_point_val = get_scalar_from_constant(zero_point)
~~~

**Comment by zhiics** [comment]
Yeah, I meant we probably don't want to put under relay root folder, somewhere else is okay to me.

**Comment by tqchen** [comment]
please suggest a resolution(move to a new folder) and act on it :)

**Comment by mbaret** [comment]
I couldn't find an appropriate folder to move this to, so I've instead inlined this function as it's only used in two places.

File: python/tvm/relay/frontend/tflite.py [hunk 1]
```diff
@@ -2219,6 +2218,20 @@ def get_expr(self, input_tensor_idx):
...
+    """ Returns scalar value from Relay constant scalar. """
+    assert isinstance(expr, _expr.Constant) and not expr.data.shape, \
+        "Expr is not a constant scalar."
+    value = expr.data.asnumpy()
+    if value.dtype == np.dtype(np.int32):
+        return int(value)
+    if value.dtype == np.dtype(np.float32):
+        return float(value)
+    assert False, "Constant expr must be float32/int32"
+    return None  # To suppress pylint
```
Comment (zhiics) [inline python/tvm/relay/frontend/tflite.py:2232]: I am okay with this type of minimal code duplication. 

Can we just do for both case?

```python
assert value.dtype == np.dtype(np.int32) or value.dtype == np.dtype(np.float32), "value must be float32/int32"
return np. asscalar(value)
```

File: python/tvm/relay/qnn/op/legalizations.py [hunk 2]
```diff
@@ -20,8 +20,8 @@
 
 import tvm
 from tvm import relay
+import numpy as np
```
Comment (roastduck) [inline python/tvm/relay/qnn/op/legalizations.py:23]: A minor problem: Lint reports `python/tvm/relay/qnn/op/legalizations.py:23: [C0411(wrong-import-order), ] third party import "import numpy as np" should be placed before "import tvm"` on my platform. I have no idea why this PR can pass CI.

Comment (mbaret) [reply python/tvm/relay/qnn/op/legalizations.py:23]: I'm not seeing this and CI doesn't appear to be failing on any other patches, can you post what command you're using to perform the linting?

Comment (roastduck) [reply python/tvm/relay/qnn/op/legalizations.py:23]: I run `./tests/scripts/task_lint.sh` directly from the repository root.

My `pylint` version:
```
pylint 2.4.4
astroid 2.3.3
Python 3.7.0 (default, Mar 12 2019, 20:10:09)
[GCC 7.3.0]
```

Comment (mbaret) [reply python/tvm/relay/qnn/op/legalizations.py:23]: If you take a look here:
https://github.com/apache/incubator-tvm/blob/063ba63a4448432a1e17ec8b793152cd8bba7c2c/docker/install/ubuntu_install_python_package.sh#L24
CI runs with pylint 1.9.4 and I imagine that's where the difference is coming from.

**Pull request merged**